client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -83,116 +88,176 @@
     // This value should be greater than the permissible value configured at the server which is by default 5 minutes.
     private static final long DEFAULT_KEEPALIVE_TIME_MINUTES = 6;
 
+    // The internal retry object to handle RPC failures.
+    private final Retry.RetryAndThrowExceptionally<StatusRuntimeException, Exception> retryConfig;
+
+    // The executor supplied by the appication to handle internal retries.
+    private final ScheduledExecutorService executor;
```

Comment:
Will clean this up. Did the bi-model approach to avoid making a lot of client side changes.
I will only use externally supplied executor and I won't need to selectively shut it down,